### PR TITLE
Use filter and insert more documents in `BenchmarkReplaceSettingsDocument`

### LINKED ISSUE #2340: Significant performance differences with MongoDB during document replacement using replaceOne()
### Versions

```
ferretdb --version
version: v0.9.4-rc.2-24-gb7bd32f
```

### What did you do?

I run logic which does following:
- Retrieves single JSON document
- Modifies single field. We do it from Go without using any MongoDB tricks
- Pushes it back using ReplaceOne to replace original document

I confirmed separately that document retrieval step performance is similar between MongoDB and FerretDB:

MongoDB:
```
time sudo ./fcli show main netflow
true

real	0m0.017s
user	0m0.008s
sys	0m0.000s

```

FerretDB:
```
time sudo ./fcli show main netflow
true

real	0m0.063s
user	0m0.002s
sys	0m0.009s

```

### What did you expect to see?

Performance on same scale with MongoDB

### What did you see instead?

Way slower performance on FerretDB with ~1s delays which clearly can be noticed:

```
time sudo ./fcli set main netflow true
Configuration updated correctly

real	0m0.822s
user	0m0.011s
sys	0m0.001s
```

MongoDB results from machine of similar configuration:
```
time sudo ./fcli set main netflow true
Configuration updated correctly

real	0m0.026s
user	0m0.012s
sys	0m0.004s
```

Both servers have NVME / SSD drives. 

We use PostgreSQL from Ubuntu repos without any configuration changes / tuning. 

### Environment

```markdown
- OS: Ubuntu 22.04
- Architecture: x86_64
```

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #2340.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/benchmarks_test.go [hunk 1]
```diff
@@ -84,18 +84,26 @@ func BenchmarkQuerySmallDocuments(b *testing.B) {
...
 	iter := shareddata.BenchmarkSettingsDocuments.NewIterator()
-	_, doc, err := iter.Next()
+	docs, err := iterator.ConsumeValuesN(iter, n)
 	iter.Close()
 
 	require.NoError(b, err)
+
+	insertDocs := make([]any, len(docs))
+	for i := range insertDocs {
+		insertDocs[i] = docs[i]
```
Comment (AlekSi) [inline integration/benchmarks_test.go:96]: We could just use a setup like other benchmarks

Comment (chilagrow) [reply integration/benchmarks_test.go]: good idea!

**Comment by chilagrow** [comment]
### bench-sqlite on this branch

```
2023-09-13T19:34:10.271+0900	INFO	setup/startup.go:90	Target system: ferretdb-sqlite (built-in).
2023-09-13T19:34:10.271+0900	INFO	setup/startup.go:103	Compat system: none, compatibility tests will be skipped.
2023-09-13T19:34:10.272+0900	INFO	debug	debug/debug.go:92	Starting debug server on http://127.0.0.1:37543/
goos: linux
goarch: amd64
pkg: github.com/FerretDB/FerretDB/integration
cpu: 12th Gen Intel(R) Core(TM) i7-1260P
BenchmarkReplaceSettingsDocument/Replace-16         	     151	  38163585 ns/op	38446824 B/op	   26750 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	     166	  36680970 ns/op	38444189 B/op	   26748 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	     169	  37578364 ns/op	38447371 B/op	   26748 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	     156	  39770445 ns/op	38448202 B/op	   26748 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	     180	  39827901 ns/op	38462472 B/op	   26758 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	     128	  45731694 ns/op	38451232 B/op	   26759 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	     138	  45367617 ns/op	38450878 B/op	   26756 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	     144	  40828568 ns/op	38442355 B/op	   26757 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	     142	  41641650 ns/op	38444676 B/op	   26757 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	     140	  41913568 ns/op	38447594 B/op	   26758 allocs/op
PASS
ok  	github.com/FerretDB/FerretDB/integration	139.140s
```
### bench-sqlite-no-pushdown
```
2023-09-13T19:30:30.620+0900	INFO	setup/startup.go:90	Target system: ferretdb-sqlite (built-in).
2023-09-13T19:30:30.620+0900	INFO	setup/startup.go:103	Compat system: none, compatibility tests will be skipped.
2023-09-13T19:30:30.620+0900	INFO	debug	debug/debug.go:92	Starting debug server on http://127.0.0.1:35483/
goos: linux
goarch: amd64
pkg: github.com/FerretDB/FerretDB/integration
cpu: 12th Gen Intel(R) Core(TM) i7-1260P
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12799343474 ns/op	9558929160 B/op	35550633 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12663396794 ns/op	9558913784 B/op	35550518 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12895047995 ns/op	9558931608 B/op	35550669 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12815387209 ns/op	9558965096 B/op	35551128 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12504455005 ns/op	9558948568 B/op	35550613 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12772391478 ns/op	9558982704 B/op	35550661 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12724095912 ns/op	9559067880 B/op	35550607 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	17386366640 ns/op	9559109688 B/op	35551977 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	16768867633 ns/op	9559183216 B/op	35552117 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	17931102349 ns/op	9559049296 B/op	35551722 allocs/op
PASS
ok  	github.com/FerretDB/FerretDB/integration	171.546s
```

### bench-sqlite on v1.9.0 release tag
```
2023-09-13T19:39:52.310+0900	INFO	setup/startup.go:90	Target system: ferretdb-sqlite (built-in).
2023-09-13T19:39:52.311+0900	INFO	setup/startup.go:103	Compat system: none, compatibility tests will be skipped.
2023-09-13T19:39:52.311+0900	INFO	debug	debug/debug.go:92	Starting debug server on http://127.0.0.1:44075/
goos: linux
goarch: amd64
pkg: github.com/FerretDB/FerretDB/integration
cpu: 12th Gen Intel(R) Core(TM) i7-1260P
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12521495663 ns/op	9557172216 B/op	35550718 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12303573404 ns/op	9557196392 B/op	35550332 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12109764424 ns/op	9557268568 B/op	35550495 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12436747066 ns/op	9557347600 B/op	35551279 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12378462453 ns/op	9557102912 B/op	35550644 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12256648758 ns/op	9556987832 B/op	35550313 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	11841048312 ns/op	9557045688 B/op	35550385 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12492299455 ns/op	9557021464 B/op	35550451 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12629146974 ns/op	9557159544 B/op	35551235 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12337122948 ns/op	9557083600 B/op	35550633 allocs/op
PASS
ok  	github.com/FerretDB/FerretDB/integration	154.463s
```

### bench-pg
```
2023-09-13T19:32:27.412+0900	INFO	setup/startup.go:90	Target system: ferretdb-pg (built-in).
2023-09-13T19:32:27.412+0900	INFO	setup/startup.go:103	Compat system: none, compatibility tests will be skipped.
2023-09-13T19:32:27.413+0900	INFO	debug	debug/debug.go:92	Starting debug server on http://127.0.0.1:36841/
goos: linux
goarch: amd64
pkg: github.com/FerretDB/FerretDB/integration
cpu: 12th Gen Intel(R) Core(TM) i7-1260P
BenchmarkReplaceSettingsDocument/Replace-16         	      21	 268314176 ns/op	38233449 B/op	   27802 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	      22	 268026482 ns/op	38234441 B/op	   27822 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	      21	 361646590 ns/op	38228023 B/op	   27764 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	      14	 404770008 ns/op	38237219 B/op	   27823 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	      14	 362692815 ns/op	38237182 B/op	   27826 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	      15	 358165936 ns/op	38231055 B/op	   27773 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	      15	 374484274 ns/op	38239071 B/op	   27825 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	      15	 365846874 ns/op	38231501 B/op	   27773 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	      15	 379915652 ns/op	38237939 B/op	   27820 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	      15	 368829267 ns/op	38229273 B/op	   27767 allocs/op
PASS
ok  	github.com/FerretDB/FerretDB/integration	149.804s
```
### bench-pg-no-pushdown
```
2023-09-13T19:35:29.948+0900	INFO	setup/startup.go:90	Target system: ferretdb-pg (built-in).
2023-09-13T19:35:29.948+0900	INFO	setup/startup.go:103	Compat system: none, compatibility tests will be skipped.
2023-09-13T19:35:29.948+0900	INFO	debug	debug/debug.go:92	Starting debug server on http://127.0.0.1:40511/
goos: linux
goarch: amd64
pkg: github.com/FerretDB/FerretDB/integration
cpu: 12th Gen Intel(R) Core(TM) i7-1260P
BenchmarkReplaceSettingsDocument/Replace-16         	       1	13390856821 ns/op	9124864384 B/op	35541845 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12974494471 ns/op	9129032856 B/op	35541732 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12994821200 ns/op	9127398640 B/op	35541362 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	13099506601 ns/op	9130604616 B/op	35542748 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12870667404 ns/op	9130279432 B/op	35541900 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12789558078 ns/op	9128440480 B/op	35541936 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	12625968525 ns/op	9125260688 B/op	35541181 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	14751800086 ns/op	9129035776 B/op	35542814 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	13788438849 ns/op	9127930912 B/op	35541541 allocs/op
BenchmarkReplaceSettingsDocument/Replace-16         	       1	13412726255 ns/op	9127747696 B/op	35542795 allocs/op
PASS
ok  	github.com/FerretDB/FerretDB/integration	221.121s
```

**Review by rumyantseva — DISMISSED** [review]
The changes look reasonable to me.

Great job with all the research and investigation!

**Review by AlekSi — APPROVED** [review]

**Pull request merged**